packages/salesforcedx-vscode-core/src/index.ts

```diff
@@ -499,6 +500,35 @@ export const activate = async (extensionContext: vscode.ExtensionContext) => {
 
   void activateTracker.markActivationStop();
   MetricsReporter.extensionPackStatus();
+
+  console.log('DOING TRACE FLAG STUFF');
+
+  const connection = await WorkspaceContext.getInstance().getConnection();
+
+  // If an expired TraceFlag exists, delete it
+  const traceFlags = await connection.tooling.query(
+    "SELECT Id, ExpirationDate FROM TraceFlag WHERE LogType = 'DEVELOPER_LOG'"
```

Comment:
that would grab any trace flag from any user.  I think the way this is currently written you'd be mixing them between users.

I think you need traceEntityId from TraceFlag to be the current user's ID
https://developer.salesforce.com/docs/atlas.en-us.api_tooling.meta/api_tooling/tooling_api_objects_traceflag.htm